tests/end_to_end/comparison/mwcc-o4p-out.c

```diff
@@ -1,16 +1,14 @@
 extern u32 global;
 
-s32 test(s32 arg0, s32 arg1, s32 arg2) {
-    MIPS2C_ERROR(unknown instruction: addic $r0, $r5, -0x1);
+f32 test(s32 arg0, s32 arg1, s32 arg2, f32 arg8) {
```

Comment:
f32? (why does this argument even exist?)

Reply:
This is a somewhat-common existing issue, related to return value heuristics. `$f1` can be the argument or the return value (or both). I could fix this in `determine_return_register` - but maybe that'd be better in a separate PR?

It looks like these tests are affected:
```
> rg -c 'return.*arg8' tests/end_to_end/
tests/end_to_end/modulo-by-power-of-two/mwcc-o4p-out.c:1
tests/end_to_end/rlwimi/mwcc-o4p-out.c:1
tests/end_to_end/division-by-power-of-two/mwcc-o4p-out.c:1
tests/end_to_end/struct/mwcc-o4p-out.c:1
tests/end_to_end/comparison/mwcc-o4p-out.c:1
tests/end_to_end/unreachable-return2/mwcc-o4p-out.c:1
tests/end_to_end/signed-conversion/mwcc-o4p-out.c:1
tests/end_to_end/load-types/mwcc-o4p-out.c:1
tests/end_to_end/loop/mwcc-o4p-out.c:3
tests/end_to_end/break/mwcc-o4p-out.c:4
tests/end_to_end/mult-by-constant/mwcc-o4p-out.c:1
```